Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc

```diff
@@ -11,66 +11,115 @@
 
 #include "Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h"
 #include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSDDDNames.h"
+#include "CondFormats/GeometryObjects/interface/PDetGeomDesc.h"
+#include "CondFormats/PPSObjects/interface/CTPPSRPAlignmentCorrectionData.h"
 
-#include "DetectorDescription/Core/interface/DDFilteredView.h"
-#include "DetectorDescription/Core/interface/DDSolid.h"
+#include "DataFormats/Math/interface/CMSUnits.h"
+#include "DetectorDescription/DDCMS/interface/DDFilteredView.h"
+#include "DetectorDescription/DDCMS/interface/DDShapes.h"
+#include "DetectorDescription/DDCMS/interface/DDSolidShapes.h"
+#include "TGeoMatrix.h"
 
-//#include "DataFormats/CTPPSAlignment/interface/RPAlignmentCorrectionData.h"
-#include "CondFormats/PPSObjects/interface/CTPPSRPAlignmentCorrectionData.h"
+#include "DataFormats/CTPPSDetId/interface/TotemRPDetId.h"
+#include "DataFormats/CTPPSDetId/interface/TotemTimingDetId.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+#include "DataFormats/Math/interface/GeantUnits.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
-using namespace std;
-
-//----------------------------------------------------------------------------------------------------
-
-DetGeomDesc::DetGeomDesc(DDFilteredView* fv)
-    : m_trans(fv->translation()),
-      m_rot(fv->rotation()),
-      m_name(((fv->logicalPart()).ddname()).name()),
-      m_params(((fv->logicalPart()).solid()).parameters()),
-      m_copy(fv->copyno()),
-      m_z(fv->geoHistory().back().absTranslation().z()),
-      m_sensorType("") {
-  std::string sensor_name = fv->geoHistory().back().logicalPart().name().fullname();
-  std::size_t found = sensor_name.find(DDD_CTPPS_PIXELS_SENSOR_NAME);
-  if (found != std::string::npos && sensor_name.substr(found - 4, 3) == DDD_CTPPS_PIXELS_SENSOR_TYPE_2x2) {
-    m_sensorType = DDD_CTPPS_PIXELS_SENSOR_TYPE_2x2;
-  }
-}
+using namespace geant_units::operators;  // Reference system is Geant length unit (mm)
```

Comment:
Is this `using` actual used below? I see fully qualified names below.

Reply:
thanks, addressed